core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -487,8 +491,9 @@ public TableMetadata removeSnapshotLogEntries(Set<Long> snapshotIds) {
         currentSnapshotId, snapshots, newSnapshotLog, addPreviousFile(file, lastUpdatedMillis));
   }
 
-  public TableMetadata buildReplacement(Schema updatedSchema, PartitionSpec updatedPartitionSpec,
-                                        Map<String, String> updatedProperties) {
+  // updatedPartitionSpec's partition field IDs should have already been refreshed
+  TableMetadata buildReplacement(Schema updatedSchema, PartitionSpec updatedPartitionSpec,
```

Comment:
Why is this no longer public?

Reply:
@rdblue  thanks for the comment.
I am thinking to expose those functionalities (`PartitionSpecUpdate` or `buildReplacement`) over `Table` and `UpdatePartitionSpec` public interface. So we can limit their access to be package/default level.

I will change them back to public in the PR and we may limit their access in the partition spec evolution PR.